From 91b883ef18368f86eebc3363042ef75c852392a5 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 14 Apr 2022 06:23:40 +0300
Subject: [PATCH] [SW-81745] gaudi3: enable erom compilation

Added pre-requisite source files to compile erom for gaudi3.

Current erom application code consists of gaudi2 specific changes
and header inclusions.

Till this is refactored to be platform agnostic, temporarily
compiling an alternate application source for gaudi3.

Change-Id: I5fc485232400b3fecfd7d215820e483f07f8d00a
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index cfdcb7a..37ca16e 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -33,6 +33,17 @@
 EROM_APP_S := $(shell ls crt0_common_erom.S)
 EROM_APP_OBJECTS := $(EROM_APP_S:.S=.o)
 
+# TODO: Revert as a part of gaudi3 erom implementation
+ifeq ($(BASE_DEV_NAME),gaudi3)
+	EROM_PCIE_SRC := $(HL_DEV_ARCH)/erom_pcie.o
+	EROM_TPM_SRC := $(HL_DEV_ARCH)/erom_tpm.o
+	EROM_RECOVERY_SRC := $(HL_DEV_ARCH)/erom_recovery.o
+else
+	EROM_PCIE_SRC := erom_pcie.o
+	EROM_TPM_SRC := erom_tpm.o
+	EROM_RECOVERY_SRC := erom_recovery.o
+endif
+
 EROM_PCIE_DEPS := \
 	$(PREBOOT_ROOT_DIR)/lib/common/pll.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/pci.o \
@@ -107,15 +118,15 @@
 	$(MAKE) erom_pcie
 	$(MAKE) erom_tpm
 
-erom_recovery: erom_recovery.o $(EROM_RECOVERY_DEPS) $(EROM_APP_OBJECTS) $(EROM_RECOVERY_LD_SCRIPT)
+erom_recovery: $(EROM_RECOVERY_SRC) $(EROM_RECOVERY_DEPS) $(EROM_APP_OBJECTS) $(EROM_RECOVERY_LD_SCRIPT)
 	@$(CC) $(LDFLAGS) -T $(EROM_RECOVERY_LD_SCRIPT) -o $@  -Wl,--start-group $< $(EROM_RECOVERY_DEPS) $(LDLIBS) $(EROM_RECOVERY_EXTRA_LIBS) -Wl,--end-group
 	$(OBJCOPY) -O binary -R .note.gnu.build-id $@ $@.bin
 
-erom_pcie: erom_pcie.o $(EROM_APP_OBJECTS) $(EROM_PCIE_DEPS) $(EROM_PCIE_LD_SCRIPT)
+erom_pcie: $(EROM_PCIE_SRC) $(EROM_APP_OBJECTS) $(EROM_PCIE_DEPS) $(EROM_PCIE_LD_SCRIPT)
 	@$(CC) $(LDFLAGS) -T $(EROM_PCIE_LD_SCRIPT) -o $@  -Wl,--start-group $< $(EROM_PCIE_DEPS) $(LDLIBS) -Wl,--end-group
 	$(OBJCOPY) -O binary -R .note.gnu.build-id $@ $@.bin
 
-erom_tpm: erom_tpm.o $(EROM_APP_OBJECTS) $(EROM_TPM_DEPS) $(EROM_TPM_LD_SCRIPT)
+erom_tpm: $(EROM_TPM_SRC) $(EROM_APP_OBJECTS) $(EROM_TPM_DEPS) $(EROM_TPM_LD_SCRIPT)
 	@$(CC) $(LDFLAGS_TPM) -T $(EROM_TPM_LD_SCRIPT) -o $@  -Wl,--start-group $< $(EROM_TPM_DEPS) $(LDLIBS) -Wl,--end-group
 	$(OBJCOPY) -O binary -R .note.gnu.build-id $@ $@.bin
 
diff --git a/arch/gaudi3/arc/erom_pcie.c b/arch/gaudi3/arc/erom_pcie.c
new file mode 100644
index 0000000..e4df6d6
--- /dev/null
+++ b/arch/gaudi3/arc/erom_pcie.c
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ * All Rights Reserved.
+ *
+ */
+
+#include <preboot.h>
+#include <d0_psoc_security_regs.h>
+
+void erom_main_function(void)
+{
+	/* Dummy magic write to SW STATUS */
+	regmap_writel(0xAABBCCDD, mmD0_PSOC_SECURITY_SW_STATUS_5);
+}
diff --git a/arch/gaudi3/arc/erom_recovery.c b/arch/gaudi3/arc/erom_recovery.c
new file mode 100644
index 0000000..3478d9d
--- /dev/null
+++ b/arch/gaudi3/arc/erom_recovery.c
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ * All Rights Reserved.
+ *
+ */
+
+#include <preboot.h>
+#include <d0_psoc_security_regs.h>
+
+void erom_main_function(void)
+{
+	/* Dummy magic write to SW STATUS */
+	regmap_writel(0xAABBCCDD, mmD0_PSOC_SECURITY_SW_STATUS_6);
+}
+
diff --git a/arch/gaudi3/arc/erom_tpm.c b/arch/gaudi3/arc/erom_tpm.c
new file mode 100644
index 0000000..3e69e71
--- /dev/null
+++ b/arch/gaudi3/arc/erom_tpm.c
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ * All Rights Reserved.
+ *
+ */
+
+#include <preboot.h>
+#include <d0_psoc_security_regs.h>
+
+void erom_main_function(void)
+{
+	/* Dummy magic write to SW STATUS */
+	regmap_writel(0xAABBCCDD, mmD0_PSOC_SECURITY_SW_STATUS_7);
+}
diff --git a/arch/gaudi3/common/pll.c b/arch/gaudi3/common/pll.c
new file mode 100644
index 0000000..4602251
--- /dev/null
+++ b/arch/gaudi3/common/pll.c
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+/*
+ * TODO: Add gaudi3 specific pll implementation
+ * Added empty source for gaudi3 eROM compilation .
+ */
diff --git a/arch/gaudi3/common/security.c b/arch/gaudi3/common/security.c
new file mode 100644
index 0000000..1c3ba90
--- /dev/null
+++ b/arch/gaudi3/common/security.c
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+/*
+ * TODO: Add gaudi3 specific security implementation
+ * Added empty source for gaudi3 eROM compilation .
+ */
diff --git a/arch/gaudi3/common/tpm2.c b/arch/gaudi3/common/tpm2.c
new file mode 100644
index 0000000..9d38b4a
--- /dev/null
+++ b/arch/gaudi3/common/tpm2.c
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+/*
+ * TODO: Add gaudi3 specific tpm source implementation
+ * Added empty source for gaudi3 eROM compilation .
+ */
diff --git a/arch/gaudi3/common/tpm_auth.c b/arch/gaudi3/common/tpm_auth.c
new file mode 100644
index 0000000..64e238b
--- /dev/null
+++ b/arch/gaudi3/common/tpm_auth.c
@@ -0,0 +1,4 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
